### PR TITLE
Add delete queue endpoints

### LINKED ISSUE #571: Downloader updateCategory and getCategories endpoints
Add `updateCategory(downloadID)` and `getAllCategories()` API endpoints for all supported downloader applications. 

This is new as we have no API paths for downloaders now.

### PR DESCRIPTION
**Pull request by davidnewhall**
Adds endpoints to delete things from the queue.

- Check spelling and wrong starr app etc.
- Default delete options?
- Closes #571 if it does what @bakerboy448 says it does.

### CODE REVIEW

File: pkg/apps/lidarr.go [hunk 1]
```diff
@@ -664,6 +665,41 @@ func lidarrGetNotifications(req *http.Request) (int, interface{}) {
...
+	idString := mux.Vars(req)["queueID"]
+	queueID, _ := strconv.ParseInt(idString, mnd.Base10, mnd.Bits64)
+
+	if !getLidarr(req).DelOK() {
+		return http.StatusLocked, ErrRateLimit
+	}
+
+	opts := &starr.QueueDeleteOpts{
+		RemoveFromClient: starr.False(),
+		BlockList:        true,
```
Comment (bakerboy448) [inline pkg/apps/lidarr.go:690]: should not default to blocklist the entry

File: pkg/apps/radarr.go [hunk 2]
```diff
@@ -1120,6 +1121,41 @@ func radarrAddNotification(req *http.Request) (int, interface{}) {
...
+	idString := mux.Vars(req)["queueID"]
+	queueID, _ := strconv.ParseInt(idString, mnd.Base10, mnd.Bits64)
+
+	if !getRadarr(req).DelOK() {
+		return http.StatusLocked, ErrRateLimit
+	}
+
+	opts := &starr.QueueDeleteOpts{
+		RemoveFromClient: starr.False(),
+		BlockList:        true,
```
Comment (bakerboy448) [inline pkg/apps/radarr.go:1146]: don't default to blocklisting

File: pkg/apps/readarr.go [hunk 3]
```diff
@@ -689,3 +690,38 @@ func readarrAddNotification(req *http.Request) (int, interface{}) {
...
+	idString := mux.Vars(req)["queueID"]
+	queueID, _ := strconv.ParseInt(idString, mnd.Base10, mnd.Bits64)
+
+	if !getReadarr(req).DelOK() {
+		return http.StatusLocked, ErrRateLimit
+	}
+
+	opts := &starr.QueueDeleteOpts{
+		RemoveFromClient: starr.False(),
+		BlockList:        true,
```
Comment (bakerboy448) [inline pkg/apps/readarr.go:716]: don't default to blocklisting

File: pkg/apps/sonarr.go [hunk 4]
```diff
@@ -1244,6 +1245,41 @@ func sonarrAddNotification(req *http.Request) (int, interface{}) {
...
+	idString := mux.Vars(req)["queueID"]
+	queueID, _ := strconv.ParseInt(idString, mnd.Base10, mnd.Bits64)
+
+	if !getSonarr(req).DelOK() {
+		return http.StatusLocked, ErrRateLimit
+	}
+
+	opts := &starr.QueueDeleteOpts{
+		RemoveFromClient: starr.False(),
+		BlockList:        true,
```
Comment (bakerboy448) [inline pkg/apps/sonarr.go:1270]: don't default to blocklisting

**Review by bakerboy448 — CHANGES_REQUESTED** [review]
given:
- RemoveFromClient: false
- BlockList: false
- SkipRedownload: false
- ChangeCategory: True

Outcome: change categories; don't search for a new release; don't add to blocklist; keep in client

**Comment by davidnewhall** [comment]
at the point of having a queue ID I do not know if it's usenet or torrent. I can add an input for the php to make the decision. Need input from @austinwbest

**Comment by austinwbest** [comment]
I honestly dont care what the default is as i'll always send it populated based on the request

**Comment by davidnewhall** [comment]
So you need to change things? What do you need to change? I need details.

**Comment by austinwbest** [comment]
> So you need to change things? What do you need to change? I need details.

You have support for all 4 parameters in starr, add support for all 4 parameters in the client

**Comment by davidnewhall** [comment]
All four queue delete options are exposed as GET parameters. They all default to `false`.

**Comment by austinwbest** [comment]
Add the parameters to the API docs for this endpoint

**Pull request merged**